src/graph/ShowExecutor.cpp

```diff
@@ -1180,6 +1183,65 @@ std::string ShowExecutor::roleToStr(nebula::cpp2::RoleType type) {
     return role;
 }
 
+std::string ShowExecutor::timestampToStr(int64_t t) {
+    std::string ret;
+    if (t == 0) {
+        return ret;
+    }
+    std::time_t tm = t;
+    char mbstr[50];
+    std::strftime(mbstr, sizeof(mbstr), "%Y-%m-%d %H:%M:%S", std::localtime(&tm));
+    return std::string(mbstr);
+}
+
+void ShowExecutor::showSessions() {
+    auto future = ectx()->getMetaClient()->listSessions();
+    auto *runner = ectx()->rctx()->runner();
+
+    auto cb = [this] (auto &&resp) {
+        if (!resp.ok()) {
+            doError(std::move(resp).status());
+            return;
+        }
+
+        auto sessionItems = std::move(resp).value();
+        resp_ = std::make_unique<cpp2::ExecutionResponse>();
+        std::vector<cpp2::RowValue> rows;
+        std::vector<std::string> header{"Host:Port", "Session ID", "Start Time", "Update Time"};
+        resp_->set_column_names(std::move(header));
+
+        std::sort(sessionItems.begin(), sessionItems.end(), [] (const auto& a, const auto& b) {
+            // sort with addr and ip
+            if (a.get_addr() == b.get_addr()) {
+                return a.get_session_id() < b.get_session_id();
+            }
+            return a.get_addr() < b.get_addr();
+        });
+
+        for (auto& item : sessionItems) {
+            std::vector<cpp2::ColumnValue> row;
+            row.resize(4);
+            row[0].set_str(item.get_addr());
+            row[1].set_integer(item.get_session_id());
+            row[2].set_str(timestampToStr(item.get_start_time()));
+            row[3].set_str(timestampToStr(item.get_update_time()));
+
+            rows.emplace_back();
```

Comment:
rows can resize sessionItems.size() first?

Reply:
I do not think so.
Because emplace_back () creates an element and inserts it at the end.